src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs

```diff
@@ -110,13 +110,28 @@ private void WriteEvent(string eventPayload)
 
         public override void LogAzureMonitorDiagnosticLogEvent(LogLevel level, string resourceId, string operationName, string category, string regionName, string properties)
         {
-            _writeEvent($"{ScriptConstants.LinuxAzureMonitorEventStreamName} {(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow}");
+            var timeStr = IsAzureMonitorTimeIsoFormatEnabled() ? DateTime.UtcNow.ToString("s") : DateTime.UtcNow.ToString();
+            _writeEvent($"{ScriptConstants.LinuxAzureMonitorEventStreamName} {(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{timeStr}");
         }
 
         public static void LogUnhandledException(Exception e)
         {
             // Pipe the unhandled exception to stdout as part of docker logs.
             Console.WriteLine($"Unhandled exception on {DateTime.UtcNow}: {e?.ToString()}");
         }
+
+        private bool IsAzureMonitorTimeIsoFormatEnabled()
+        {
+            string enabledString = Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureMonitorTimeIsoFormatEnabled);
```

Comment:
Please use `IOptions` pattern and not reading env variables directly.

Reply:
@vivekjilla - here is an example - https://github.com/Azure/azure-functions-host/blob/ba3e9ef990a9a5ebacaf47bbc3a71ed5f661e8ea/src/WebJobs.Script.WebHost/Diagnostics/LinuxAppServiceEventGenerator.cs#L89

Reply:
I see we have `IOptions<FunctionsHostingConfigOptions>` which seems to read the config from `FunctionsHostingConfig` section. Have few questions here:
1. Can customers set this value in FunctionsHostingConfig, and if so, how? or otherwise
2. I believe the app settings set by the customers can be accessed via environment variables. Do we have any example of IOptions which gets populated based on environment variables.

Or if I'm completely off the direction suggested here, I can catch up with @jviau offline.

Reply:
@vivekjilla, IOptions pattern reads from IConfiguration (and other custom ways). And IConfiguration is built from many sources - including environment variables. So, app settings (which are env vars at runtime) are available in IOptions and IConfiguration pattern.

As for which options object to use - you can see if there is a fitting one, or you can always add your own. So, you don't have to specifically use the any existing IOptions object if they are not fitting and instead create a new one.

That is the purpose of options pattern: use an options POCO which is specific to the _usage_ of the values and abstract away how the values are populated (in the options pipeline).